util/blockdevice/util.go

```diff
@@ -88,7 +88,31 @@ func GetCapacityOrError(obj unstructured.Unstructured) (resource.Quantity, error
 			errors.Errorf("Can not get capacity: Expected kind %q got %q",
 				types.KindBlockDevice, obj.GetKind())
 	}
-	return unstruct.GetQuantityOrError(&obj, "spec", "capacity", "storage")
+	return unstruct.GetInt64AsQuantityOrError(&obj, "spec", "capacity", "storage")
+}
+
+// GetLogicalSectorSizeOrError returns LogicalSectorSize of a block device in
+// resource.Quantity format If value if not found or for an invalid capacity
+// then it returns an error
+func GetLogicalSectorSizeOrError(obj unstructured.Unstructured) (resource.Quantity, error) {
+	if obj.GetKind() != string(types.KindBlockDevice) {
+		return resource.Quantity{},
+			errors.Errorf("Can not get capacity: Expected kind %q got %q",
+				types.KindBlockDevice, obj.GetKind())
+	}
+	return unstruct.GetInt64AsQuantityOrError(&obj, "spec", "capacity", "logicalSectorSize")
+}
+
+// GetPhysicalSectorSizeOrError returns PhysicalSectorSize of a block device
+// in resource.Quantity format If value if not found or for an invalid capacity
+// then it returns an error
+func GetPhysicalSectorSizeOrError(obj unstructured.Unstructured) (resource.Quantity, error) {
+	if obj.GetKind() != string(types.KindBlockDevice) {
```

Comment:
If obj itself is null, this will panic

Reply:
I am not using pointer. It will not be nil.
if the object inside unstructured.Unstructured{} is nil then that case is handled here
https://github.com/kubernetes/kubernetes/blob/4bb8df4464f44090ec1a595fbcaa60288e1e0d82/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go#L104